apps/veil/src/pages/tournament/api/use-epoch-results.ts

```diff
@@ -18,5 +31,47 @@ export const useEpochResults = (
 
   useRefetchOnNewBlock(name, query, disabled);
 
-  return query;
+  // collect all the gauges for the current epoch by the asset denom
+  const gaugeMapByDenom = useMemo(
+    () =>
+      (query.data?.data ?? []).reduce<Map<string, MappedGauge>>((accum, current) => {
+        accum.set(current.asset.base, current);
+        return accum;
+      }, new Map()),
+    [query.data?.data],
+  );
+
+  // filters assets from the registry that are IBC-assets and match the search query
+  const filteredAssets = useMemo<Metadata[]>(() => {
+    return assets.filter(asset => {
+      return (
+        assetPatterns.ibc.matches(asset.base) &&
+        (asset.symbol.toLowerCase().includes(search.toLowerCase()) ||
+          asset.description.toLowerCase().includes(search.toLowerCase()))
+      );
+    });
+  }, [assets, search]);
```

Comment:
If we don't display assets having received no votes, then this code can be removed as well.

Reply:
we need this to display asset with no votes and improve overall UX – users need to see options to vote for

protocol might filted for ibc assets, but the registry doesn't

Reply:
Yeah, that's a good point about needing to see the options with no votes. In practice the only asset this will filter out is UM, but it's fine to have this logic.